mtags/src/main/scala-3/scala/meta/internal/pc/SelectionRangeProvider.scala

```diff
@@ -103,6 +98,34 @@ object SelectionRangeProvider:
   import scala.meta.*
   import dotty.tools.dotc.ast.tpd
 
+  /** Given a tree, create a seq of [[SelectionRange]]s corresponding to that tree. */
+  private def selectionRangesFromTree(pos: SourcePosition)(tree: tpd.Tree)(using Context) =
+    def toSelectionRange(srcPos: SourcePosition) =
+      val selectionRange = new SelectionRange()
+      selectionRange.setRange(srcPos.toLsp)
+      selectionRange
+
+    val treeSelectionRange = toSelectionRange(tree.sourcePos)
+
+    tree match
+      case tpd.DefDef(name, paramss, tpt, rhs) =>
+        // If source position is within a parameter list, add a selection range covering that whole list.
+        val selectedParams =
+          paramss
+            .iterator
+            .flatMap{ // parameter list to a sourcePosition covering the whole list
+              case Seq(param) => Some(param.sourcePos)
+              case params @ Seq(head, tail*) =>
```

Comment:
```suggestion
              case params @ Seq(head, tail*) if tail.size > 1=>
```
just to be sure this doesn't break if anything is moved around

Reply:
I don't disagree, but making this change emits a non-exhaustive match warning.

Reply:
ach yes, you would need to default with `_` to None.

Reply:
Done ...